src/allotropy/parsers/tecan_magellan/tecan_magellan_structure.py

```diff
@@ -0,0 +1,204 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from io import StringIO
+from pathlib import Path
+import re
+
+import pandas as pd
+
+from allotropy.allotrope.schema_mappers.adm.plate_reader.rec._2024._06.plate_reader import (
+    Measurement,
+    MeasurementGroup,
+    MeasurementType,
+    Metadata,
+)
+from allotropy.exceptions import AllotropeConversionError
+from allotropy.parsers.constants import NOT_APPLICABLE
+from allotropy.parsers.lines_reader import LinesReader
+from allotropy.parsers.tecan_magellan import constants
+from allotropy.parsers.utils.pandas import df_to_series_data, read_csv, SeriesData
+from allotropy.parsers.utils.uuids import random_uuid_str
+from allotropy.parsers.utils.values import assert_not_none, try_float
+
+
+def get_measurement_type(measurement_mode: str) -> MeasurementType:
+    if measurement_mode.lower() == "absorbance":
+        return MeasurementType.ULTRAVIOLET_ABSORBANCE
+
+    msg = f"Only Absorbance measurements are supported as this time. Got {measurement_mode}."
+    raise AllotropeConversionError(msg)
+
+
+def parse_settings_lines(lines: list[str]) -> SeriesData:
```

Comment:
Could we do this in the reader class? So the reader would end up with a metadata or settings SeriesData memeber.

Reply:
Discussed offline - missed that you are parsing settings lines out of multiple sections.